service/src/test/java/uk/nhs/adaptors/gp2gp/uat/EhrExtractUATTest.java

```diff
@@ -0,0 +1,165 @@
+    @Mock
+    private CodeableConceptCdMapper codeableConceptCdMapper;
+
+    private EhrExtractMapper ehrExtractMapper;
+    private MessageContext messageContext;
+
+    @BeforeEach
+    public void setUp() {
+        getGpcStructuredTaskDefinition = GetGpcStructuredTaskDefinition.builder()
+            .nhsNumber(TEST_NHS_NUMBER)
+            .conversationId(TEST_CONVERSATION_ID)
+            .requestId(TEST_REQUEST_ID)
+            .fromOdsCode(TEST_FROM_ODS_CODE)
+            .toOdsCode(TEST_TO_ODS_CODE)
+            .build();
+
+        when(randomIdGeneratorService.createNewId()).thenReturn(TEST_ID_1, TEST_ID_2, TEST_ID_3);
+        when(timestampService.now()).thenReturn(Instant.parse(TEST_DATE_TIME));
+        messageContext = new MessageContext(randomIdGeneratorService);
+        EncounterComponentsMapper encounterComponentsMapper = new EncounterComponentsMapper(
+            messageContext,
+            new DiaryPlanStatementMapper(messageContext, codeableConceptCdMapper),
+            new NarrativeStatementMapper(messageContext),
+            new ObservationStatementMapper(
+                messageContext,
+                new StructuredObservationValueMapper(),
+                new PertinentInformationObservationValueMapper(),
+                codeableConceptCdMapper
+            ),
+            new ImmunizationObservationStatementMapper(messageContext, codeableConceptCdMapper),
+            new ConditionLinkSetMapper(messageContext, randomIdGeneratorService, codeableConceptCdMapper),
+            new BloodPressureMapper(
+                messageContext, randomIdGeneratorService, new StructuredObservationValueMapper(), codeableConceptCdMapper)
+        );
+
+        ehrExtractMapper = new EhrExtractMapper(randomIdGeneratorService,
+            timestampService,
+            new EncounterMapper(messageContext, encounterComponentsMapper));
+    }
+
+    @AfterEach
+    public void tearDown() {
+        messageContext.resetMessageContext();
+    }
+
+    @ParameterizedTest
+    @MethodSource("testValueFilePaths")
+    public void When_MappingProperJsonRequestBody_Expect_ProperXmlOutput(String inputJson, String expectedOutputXml) throws IOException {
+        String expectedJsonToXmlContent = ResourceTestFileUtils.getFileContent(OUTPUT_PATH + FILES_PREFIX + expectedOutputXml);
+        String inputJsonFileContent = ResourceTestFileUtils.getFileContent(INPUT_PATH + FILES_PREFIX + inputJson);
+        Bundle bundle = new FhirParseService().parseResource(inputJsonFileContent, Bundle.class);
+        messageContext.initialize(bundle);
+
+        EhrExtractTemplateParameters ehrExtractTemplateParameters = ehrExtractMapper.mapBundleToEhrFhirExtractParams(
+            getGpcStructuredTaskDefinition,
+            bundle);
+        String outputXml = ehrExtractMapper.mapEhrExtractToXml(ehrExtractTemplateParameters);
+
+        assertThat(outputXml).isEqualToIgnoringWhitespace(expectedJsonToXmlContent);
```

Comment:
I think we should not be using isEqualToIgnoringWhitespace 
(as per NIAD-1137)

Reply:
Here is a fix cdcaf2f8f3a798de96365af1141d7059f93ade85.
Thanks to @Broom94 for the help! @z-uddin if you are wondering how it has been done because we tried to do it yesterday with you and just copying the output from the terminal didn't work. The way was to remove the XML file, create a new one, copy the output from the terminal in a VSCode (not in IntelliJ) and add the empty line at the end. Only following exactly this pattern has worked. I've tried to it with already existing files and it didn't work. Very weird issue, not really sure why it works in the way it works but now it has been fixed. Thank you!